blog/BLOG.md

```diff
@@ -0,0 +1,99 @@
+
+# Interpretting the commands to present the actual terminal
+The screens follow each other, so from previous screen there might be characters left to the next screen. if I want to know what is printed in screen 401, I need to go through all the screens from 1 to 400 in case they leave any output to be printed in screen 401. All the commands have an output, which is the actual printable string. Some commands have different variables like MoveArrowCommand has booleans up, down, left and right and CursorMoveCommand has a row and col to tell where to move the cursor before output. Interpretting is just looping the commands of a screen. At the end we "print out" the output with parseOutputToTerminal-function.
+```
+foreach ($commands as $command) {
+    $commClass = get_class($command);
+    switch($commClass)
+    {
+        case ClearScreenCommand::class:
+            $this->clearConsole();
+            break;
+        case BackspaceCommand::class:
+            $this->cursorCol--;
+            break;
+        case NewlineCommand::class:
+            $this->cursorCol = self::COLUMN_BEGINNING;
+            $this->cursorRow++;
+            $this->parseOutputToTerminal($command->getOutput());
+            break;
+        case CarriageReturnCommand::class:
+            $this->cursorCol = self::COLUMN_BEGINNING;
+            $this->parseOutputToTerminal($command->getOutput());
+            break;
+        case CursorMoveCommand::class:
+            $this->cursorRow = $command->row;
+            $this->cursorCol = $command->col;
+            $this->parseOutputToTerminal($command->getOutput());
+        ...
+```
+
+The end result of this looping is a array that is filled with terminal rows. Then we can just output them into anything.
+
+Like a text file
+```
+$lastLine = max(array_keys($this->console));
+$data = '';
+for ($i = 0;$i <= $lastLine;$i++) {
+  if (isset($this->console[$i])) {
+    $data .= $this->console[$i]->output;
+  }
+  $data .= PHP_EOL;
+}
+file_put_contents("screen.txt", $data);
+```
+or to a gif
+```
+$lastLine = max(array_keys($this->console));
+$im = imagecreate($this->imageWidth, $this->imageHeight);
+$this->setBackgroundColor($im);
+$textcolor = $this->getForegroundColor($im);
+for ($i = 1;$i <= lastLine;$i++) {
+    if (isset($this->console[$i])) {
+        $x = $this->margin;
+        $y = $i * $this->fontHeight + $this->margin;
+        $text = $this->console[$i]->output;
+        imagestring($im, $this->font, $x, $y, $text, $textcolor);
+    }
+}
+imagegif($im, $filename);
+```
+Nice. After doing this, I have 6415 individual gif files. How do I combine them into a single animated gif?
```

Comment:
Link to part 2 vois olla suoraan tän lauseen perään. "See part 2 to find out" tms.